weldx/asdf/file.py

```diff
@@ -300,6 +312,21 @@ def __init__(
         # initialize protected key interface.
         super().__init__(protected_keys=_PROTECTED_KEYS, data=self._asdf_handle.tree)
 
+    @contextmanager
+    def _config_context(self, **kwargs):
+        # Temporarily set (default) options in asdf.config_context. This is useful
+        # during writing/updating data.
+        if (
+            "array_inline_threshold" not in kwargs
+            or kwargs["array_inline_threshold"] is None
+        ):
+            kwargs["array_inline_threshold"] = self._array_inline_threshold
+
+        with config_context() as config:
+            for k, v in kwargs.items():
+                setattr(config, k, v)
+            yield
+
```

Comment:
If I understand this correctly all (internal) `WeldxFile` operations should now use this custom context? I think this is a very good solution and more in line with the new asdf API 👍

Reply:
Yes, currently it is only used where really needed. That includes write operations. But I guess later on, we will wrap everything in this context.